tools/tplgtool2.py

```diff
@@ -470,7 +470,7 @@ def __init__(self):
             "mask" / Int32ul,
             "count" / Int32ul,
             "texts" / Array(self._num_texts, String(self._id_name_maxlen, "ascii")),
-            "values" / Array(self._num_texts * self._id_name_maxlen / 4, Int32ul),
+            "values" / Array(int(self._num_texts * self._id_name_maxlen / 4), Int32ul),
```

Comment:
I think it's safer and clearer to avoid floats and conversions entirely:
```suggestion
            "values" / Array(self._num_texts * self._id_name_maxlen // 4, Int32ul),
```

Or even maybe better:
```suggestion
            self._id_name_maxint32 = 11
            ....
            "values" / Array(self._num_texts * self._id_name_maxint32, Int32ul),
```

Reply:
what is the problem with conversion?

Reply:
Every floating point operation should immediately raise the "What is the rounding error this time?" red flag. Even when there's none, proving that there is none is a waste of mental energy. It's just weird to perform pure-integer arithmetic using floats; I bet most people would object to that in C.

```
python3
>>> 0.1+0.2
0.30000000000000004
>>>
```

It's also weird to repeatedly divide 44 by 4 every time.

Reply:
fair enough. fixed now